docs/CreatingClientInstance.md

```diff
@@ -0,0 +1,68 @@
+# Creating Client Instance
+
+Initialization of the Client can be done in one of below two ways
+
+## 1. Create With ClientOptions [Recommended]
+
+In order to instantiate a Client object, one has to pass in the `authProvider` or `middleware` chain in [ClientOptions](../src/IClientOptions.ts).
+
+### 1. Default Middleware chain
```

Comment:
"1. Default Middleware chain" becomes "Option A. Default Middleware chain". It is confusing to have a numbered heading that matches with the simialrly numbered subheading